grammar.js

```diff
@@ -1,3 +1,9 @@
+const PREC = {
```

Comment:
Can you use tree-sitter's partial precedence feature for this?

Reply:
I tried this one. And it works when distinguishing `block_comment_text` and `pluscal_algorithm`. However it fails to distinguish `comment` and `block_comment`. Would be happy to find a way to use partial order here

Reply:
If this works but partial order doesn't, then either means there's a bug in partial precedence or global order is important in the parsing of comments (since rules without precedence have default global precedence 0).

Reply:
Opened a [discussion](https://github.com/tree-sitter/tree-sitter/discussions/1598)

Reply:
Okay, we can keep it as-is until that discussion is resolved then change it later as necessary.